contribs/gmf/src/directives/search.js

```diff
@@ -0,0 +1,280 @@
+    controllerAs: 'ctrl',
+    template:
+        '<input type="text" placeholder="search…" ' +
+        'ngeo-search="ctrl.options" ' +
+        'ngeo-search-datasets="ctrl.datasets" ' +
+        'ngeo-search-listeners="ctrl.listeners">',
+    link:
+        /**
+         * @param {angular.Scope} scope Scope.
+         * @param {angular.JQLite} element Element.
+         * @param {angular.Attributes} attrs Atttributes.
+         */
+        function(scope, element, attrs) {
+          // Empty the search field on focus and blur.
+          element.find('input').on('focus blur', function() {
+            $(this).val('');
+          });
+        }
+  };
+};
+
+gmfModule.directive('gmfSearch', gmf.searchDirective);
+
+
+
+/**
+ * @constructor
+ * @param {angular.Scope} $scope The directive's scope.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {ngeo.CreateGeoJSONBloodhound} ngeoCreateGeoJSONBloodhound The ngeo
+ * create GeoJSON Bloodhound service.
+ * @ngInject
+ * @export
+ */
+gmf.SearchController = function($scope, $compile,
+    ngeoCreateGeoJSONBloodhound) {
+
+  /**
+   * @type {angular.Scope}
+   * @private
+   */
+  this.scope_ = $scope;
+
+  /**
+   * @type {angular.$compile}
+   * @private
+   */
+  this.compile_ = $compile;
+
+  /**
+   * @type {ngeo.CreateGeoJSONBloodhound}
+   * @private
+   */
+  this.ngeoCreateGeoJSONBloodhound_ = ngeoCreateGeoJSONBloodhound;
+
+  var map = this.scope_['getMapFn']();
+  goog.asserts.assertInstanceof(map, ol.Map);
+
+  /**
+   * @type {!ol.Map}
+   * @private
+   */
+  this.map_ = map;
+
+  /**
+   * @type {ol.layer.Vector}
+   * @private
+   */
+  this.vectorLayer_ = this.createVectorLayer_();
+
+  var datasources = this.scope_['getDatasourcesFn']();
+  goog.asserts.assertArray(datasources);
+
+  /**
+   * @type {Array.<gmfx.SearchDirectiveDatasource>}
+   * @private
+   */
+  this.datasources_ = datasources;
+
+  /**
+   * @type {TypeaheadOptions}
+   * @export
+   */
+  this.options = /** @type {TypeaheadOptions} */ ({
+    highlight: true,
+    hint: undefined,
+    minLength: undefined
+  });
+
+  /**
+   * @type {Array.<TypeaheadDataset>}
+   * @export
+   */
+  this.datasets = [];
+
+  for (var i = 0; i < this.datasources_.length; i++) {
+    var datasource = this.datasources_[i];
+    var groupByProperties = datasource.groupByProperties || [];
+    var filter;
+
+    do {
+      var title = datasource.title;
+
+      if (groupByProperties[0]) {
+        //Add an optional filter function to keep objects only from one
+        //"layername" from a GMF's fulltextsearch service.
+        filter = this.filterLayername_(groupByProperties[0]);
+        title = title + groupByProperties[0];
+        groupByProperties.shift();
+      } else {
+        filter = undefined;
+      }
+
+      this.datasets.push(this.createDataset_({
+        'title': title,
+        'filter': filter,
+        'projection': datasource.projection,
+        'url': datasource.url
+      }));
```

Comment:
You shouldn't use quotes for the object passed to `createDataset_`. I also realize that you use a `filter` property here that is not defined in the `gmfx.SearchDirectiveDatasource`. This requires more rigor in my opinion. Maybe you could just pass the filter with an extra argument to `createDataset_`.